hail/src/test/scala/is/hail/io/compress/BGzipCodecSuite.scala

```diff
@@ -114,4 +115,57 @@ class BGzipCodecSuite extends SparkSuite {
     }
     p.check()
   }
+
+  @Test def testVirtualSeek() {
+    sc.hadoopConfiguration.setLong("mapreduce.input.fileinputformat.split.minsize", 1L)
+    // real offsets of the start of each block
+    val blockStarts = Array[Long](14653, 28034, 40231, 55611, 69140, 82949, 96438, 110192, 121461, 133703,
+      146664, 158711, 171239, 181362)
+    // offsets into the uncompressed file
+    val uncompBlockStarts = Array[Long](65280, 130560, 195840, 261120, 326400, 391680, 456960, 522240,
+      587520, 652800, 718080, 783360, 848640, 913920)
+
+    val blockPointerOffset = 16 // magic number for creating virtual file pointers into bgzipped files
+
+    val uncompPath = "src/test/resources/sample.vcf"
+    val compPath = "src/test/resources/sample.vcf.gz"
+
+    val uncompHPath = new hd.fs.Path(uncompPath)
+    val compHPath = new hd.fs.Path(compPath)
+
+    val fs = uncompHPath.getFileSystem(hadoopConf)
+    val compIS = fs.open(compHPath)
+    val uncompIS = fs.open(uncompHPath)
+
+    val decompIS = new BGzipInputStream(compIS)
+
+    for ((cOff, uOff) <- blockStarts.zip(uncompBlockStarts);
+         extra <- Seq(0, 50)) {
+      val decompData = new Array[Byte](100)
+      val uncompData = new Array[Byte](100)
+      val vptr = cOff << blockPointerOffset | extra
+
+      decompIS.virtualSeek(vptr)
+      uncompIS.seek(uOff + extra)
+
+      val decompRead = decompIS.read(decompData)
+      val uncompRead = uncompIS.read(uncompData)
+
+      Assert.assertEquals(decompRead, uncompRead, s"""compressed offset: ${ cOff }
+        |decomp bytes read: ${ decompRead }
+        |uncomp bytes read: ${ uncompRead }\n""".stripMargin)
+      Assert.assertEquals(decompData, uncompData, s"data differs for compressed offset: ${ cOff }")
+    }
+
+    try {
+      val vptr = blockStarts(3) << blockPointerOffset | uncompBlockStarts(0)
```

Comment:
This feels like a semantic error (or'ing in the block start instead of a block offset).  Can you include the block sizes and use the block size (or something greater?)